lib/Command/UpdateAccount.php

```diff
@@ -0,0 +1,164 @@
+        public const ARGUMENT_SMTP_HOST = 'smtp-host';
+        public const ARGUMENT_SMTP_PORT = 'smtp-port';
+        public const ARGUMENT_SMTP_SSL_MODE = 'smtp-ssl-mode';
+        public const ARGUMENT_SMTP_USER = 'smtp-user';
+        public const ARGUMENT_SMTP_PASSWORD = 'smtp-password';
+
+	/** @var mapper */
+        private $mapper;
+
+	/** @var ICrypto */
+	private $crypto;
+
+	public function __construct(MailAccountMapper $mapper, ICrypto $crypto) {
+		parent::__construct();
+
+                $this->mapper = $mapper;
+		$this->crypto = $crypto;
+	}
+
+	/**
+	 * @return void
+	 */
+	protected function configure() {
+		$this->setName('mail:account:update');
+		$this->setDescription('Update a user\'s IMAP account(s)');
+		$this->addArgument(self::ARGUMENT_USER_ID, InputArgument::REQUIRED);
+                $this->addArgument(self::ARGUMENT_EMAIL, InputArgument::REQUIRED);
+
+
+                $this->addOption(self::ARGUMENT_IMAP_HOST,'', InputOption::VALUE_REQUIRED);
+                $this->addOption(self::ARGUMENT_IMAP_PORT,'', InputOption::VALUE_REQUIRED);
+                $this->addOption(self::ARGUMENT_IMAP_SSL_MODE,'', InputOption::VALUE_REQUIRED);
+                $this->addOption(self::ARGUMENT_IMAP_USER,'', InputOption::VALUE_REQUIRED);
+                $this->addOption(self::ARGUMENT_IMAP_PASSWORD,'', InputOption::VALUE_REQUIRED);
+               
+                $this->addOption(self::ARGUMENT_SMTP_HOST,'', InputOption::VALUE_REQUIRED);
+                $this->addOption(self::ARGUMENT_SMTP_PORT,'', InputOption::VALUE_REQUIRED);
+                $this->addOption(self::ARGUMENT_SMTP_SSL_MODE,'', InputOption::VALUE_REQUIRED);
+                $this->addOption(self::ARGUMENT_SMTP_USER,'', InputOption::VALUE_REQUIRED);
+                $this->addOption(self::ARGUMENT_SMTP_PASSWORD,'',InputOption::VALUE_REQUIRED);
+
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+
+                $userId = $input->getArgument(self::ARGUMENT_USER_ID);
+                $email = $input->getArgument(self::ARGUMENT_EMAIL);
+
+
+                $imapHost = $input->getOption(self::ARGUMENT_IMAP_HOST);
+                $imapPort = $input->getOption(self::ARGUMENT_IMAP_PORT);
+                $imapSslMode = $input->getOption(self::ARGUMENT_IMAP_SSL_MODE);
+                $imapUser = $input->getOption(self::ARGUMENT_IMAP_USER);
+                $imapPassword = $input->getOption(self::ARGUMENT_IMAP_PASSWORD);
+
+                $smtpHost = $input->getOption(self::ARGUMENT_SMTP_HOST);
+                $smtpPort = $input->getOption(self::ARGUMENT_SMTP_PORT);
+                $smtpSslMode = $input->getOption(self::ARGUMENT_SMTP_SSL_MODE);
+                $smtpUser = $input->getOption(self::ARGUMENT_SMTP_USER);
+                $smtpPassword = $input->getOption(self::ARGUMENT_SMTP_PASSWORD);
+
+                $accounts = $this->mapper->findByUserId($userId);
```

Comment:
```suggestion
                $accounts = $this->mapper->findByUserIdAndEmail($userId, $email);
```

Please add a method to the mapper where you query the db with both params directly. Then you can avoid the foreach loop.

Something like this should suffice:

```php
	/**
	 * Finds an mail account by user id and email address
	 *
	 * @return MailAccount
	 * @throws DoesNotExistException
	 */
	public function findByUserIdAndEmail(string $userId, string $email): MailAccount {
		$qb = $this->db->getQueryBuilder();
		$query = $qb
			->select('*')
			->from($this->getTableName())
			->where($qb->expr()->eq('user_id', $qb->createNamedParameter($userId, IQueryBuilder::PARAM_STR), IQueryBuilder::PARAM_STR))
			->andWhere($qb->expr()->eq('email', $qb->createNamedParameter($email, IQueryBuilder::PARAM_STR), IQueryBuilder::PARAM_STR));

		return $this->findEntity($query);
	}
```

Reply:
Great!. Thank you for the piece of code